server/get_payload.go

```diff
@@ -235,20 +236,29 @@ func (m *BoostService) innerGetPayload(log *logrus.Entry, signedBlindedBeaconBlo
 				RecordRelayStatusCode(strconv.Itoa(statusCode), endpoint, relay.String())
 				// Check that the response was successful
 
-				// If the relay does not support V2 API, retry with V1 API
-				// we can fallback to V1 API if the status code returned >= 400. There is no harm
-				// falling back to the V1 API, falling back to the V1 API in the case of any error
-				// can be beneficial to the proposer to avoid a missed slot.
-				if resp.StatusCode >= http.StatusBadRequest && url == relay.GetURI(params.PathGetPayloadV2) {
-					log.Warn("relay may not support V2 API, Retrying with V1 API")
-					// retry with v1 api
-					url = relay.GetURI(params.PathGetPayload)
-					versionToUse = GetPayloadV1
-					return nil, errRetryWithV1API
-				}
+				// If the response status code doesn't match expected, read error body once
 				if resp.StatusCode != statusCode {
-					err = fmt.Errorf("%w: %d", errHTTPErrorResponse, resp.StatusCode)
-					log.WithError(err).Warn("error status code")
+					errorBody, err := io.ReadAll(resp.Body)
+					if err != nil {
+						log.WithError(err).Warn("error reading error body")
+						return nil, err
+					}
+					errorBodyStr := string(errorBody)
+					log.WithField("errorBody", errorBodyStr).Warnf("unexpected status code %d", resp.StatusCode)
+
+					// If the relay does not support V2 API, retry with V1 API
+					// we can fallback to V1 API if the status code returned >= 400. There is no harm
+					// falling back to the V1 API, falling back to the V1 API in the case of any error
+					// can be beneficial to the proposer to avoid a missed slot.
+					if resp.StatusCode >= http.StatusBadRequest && url == relay.GetURI(params.PathGetPayloadV2) {
+						log.Warn("relay may not support V2 API, Retrying with V1 API")
```

Comment:
The capitalized "Retrying" word should be lowercase.
```suggestion
						log.Warn("relay may not support V2 endpoint, retrying with V1 endpoint")
```

Reply:
Or this if you prefer:
```suggestion
						log.Warn("relay may not support getPayloadV2, retrying with getPayloadV1")
```